### PR TITLE
Create basic_modal.py

### PR DESCRIPTION
**Pull request by dannynotsmart**
Simple example used for showing how modals are used.

## Summary

This example serves to show how you would use modals in your code. I chose to make it be a suggestion bot because majority of the users I feel will either go with mod applications, reports, or suggestions.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: examples/suggestions_modal.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            'Your suggestion has been sent successfully!', ephemeral=True
+        )
+
+
+client = SuggestionsBot()
+
+# We create a slash command that opens the modal
+@client.tree.command(
+    guild=discord.Object(id=my_guild_id),
+    name='suggest'
```
Comment (pikaninja) [inline examples/suggestions_modal.py:74]: this isn't neccesary

Comment (dannynotsmart) [reply examples/suggestions_modal.py:74]: ah my bad

Comment (dannynotsmart) [reply examples/suggestions_modal.py]: i have updated the file

File: examples/basic_modal.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    reasoning = discord.ui.TextInput(
+        label='Provide a reasoning for this suggestion.',
+        min_length=10,
+        max_length=100,
+        style=discord.TextStyle.paragraph,
+    )
+
+    async def on_submit(self, interaction: discord.Interaction):
+
```
Comment (StockerMC) [inline examples/basic_modal.py:40]:
```suggestion
```

Comment (dannynotsmart) [reply examples/basic_modal.py:40]: Oops!

**Comment by Rapptz** [comment]
Hello.

This example seems like a duplicate effort of #7521. Since that one was first, I think I'll close this one in favour of the other one. Sorry about this.

**Pull request closed without merging**